src/main/java/org/minions/devfund/damianvp/battleship/Ship.java

```diff
@@ -0,0 +1,231 @@
+package org.minions.devfund.damianvp.battleship;
+
+/**
+ * Ship class.
+ */
+public abstract class Ship {
+    private int bowRow;
+    private int bowColumn;
+    private int length;
+    private boolean horizontal;
+    private boolean[] hit;
+
+    /**
+     * Method to get BowRow of current class.
+     * @return int type, bowRow.
+     */
+    public int getBowRow() {
+        return bowRow;
+    }
+
+    /**
+     * Method to get bowColumn.
+     * @return int type, bowColumn.
+     */
+    public int getBowColumn() {
+        return bowColumn;
+    }
+
+    /**
+     * Method to get the length of ship.
+     * @return int type, length of ship.
+     */
+    public int getLength() {
+        return length;
+    }
+
+    /**
+     * This method verifies if position is horizontal.
+     * @return true if ship is placed horizontally.
+     *         false if ship is placed vertically.
+     */
+    public boolean isHorizontal() {
+        return horizontal;
+    }
+
+    /**
+     * This method gets the array hit.
+     * @return boolean array, hit.
+     */
+    protected boolean[] getHit() {
+        return hit;
+    }
+
+    /**
+     * This method sets the bow row of ship.
+     * @param bowRow int, row position.
+     */
+    public void setBowRow(int bowRow) {
+        this.bowRow = bowRow;
+    }
+
+    /**
+     * This method sets the bow column of ship.
+     * @param bowColumn int, column position.
+     */
+    public void setBowColumn(int bowColumn) {
+        this.bowColumn = bowColumn;
+    }
+
+    /**
+     * This method sets the length of ship.
+     * @param length int, length of ship.
+     */
+    public void setLength(int length) {
+        this.length = length;
+    }
+
+    /**
+     * This method sets if ship is horizontal or vertical.
+     * @param horizontal bolean, (true if ship is horizontal, false if ship is vertical).
+     */
+    public void setHorizontal(boolean horizontal) {
+        this.horizontal = horizontal;
+    }
+
+    /**
+     * This method set the hit.
+     * @param hit boolean array.
+     */
+    public void setHit(final boolean[] hit) {
+        this.hit = hit.clone();
+    }
+
+    /**
+     * abstract method.
+     * @return the ship type.
+     */
+    abstract String getShipType();
+
+    /**
+     * This method verifies if position are ok to place a ship.
+     * @param row int row position.
+     * @param column int column position.
+     * @param horizontal boolean, (true if ship is horizontal, false if ship is vertical).
+     * @param ocean Ocean type.
+     * @return true if positions are ok to place a ship.
+     *         false if positions does not ok to place a ship.
+     */
+    boolean okToPlaceShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+        int rowLimitA;
+        int rowLimitB;
+        int columnLimitA;
+        int columnLimitB;
+        final int length = ocean.getShips().length;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "length" which hides the field declared at line 9. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)